colossus/src/main/scala/colossus/protocols/http/HttpResponseParser.scala

```diff
@@ -15,13 +15,14 @@ object HttpResponseParser  {
 
   import HttpBody._
 
-  protected def staticBody(dechunk: Boolean): Parser[HttpResponse] = head |> {parsedHead =>
+  protected def staticBody(dechunk: Boolean): Parser[HttpResponse] = head |> { parsedHead =>
+    val contentType = parsedHead.headers.contentType
     parsedHead.headers.transferEncoding match {
       case TransferEncoding.Identity => parsedHead.headers.contentLength match {
         case Some(0)  => const(HttpResponse(parsedHead, HttpBody.NoBody))
-        case Some(n)  => bytes(n) >> {body => HttpResponse(parsedHead, new HttpBody(body))}
+        case Some(n)  => bytes(n) >> { body => HttpResponse(parsedHead, new HttpBody(body, contentType)) }
         case None if (parsedHead.code.isInstanceOf[NoBodyCode]) => const(HttpResponse(parsedHead, HttpBody.NoBody))
-        case None     => bytesUntilEOS >> {body => HttpResponse(parsedHead, body)}
+        case None     => bytesUntilEOS >> { body => HttpResponse(parsedHead, body) }
```

Comment:
Actually, `bytesUntilEOS` should be returning an `Array[Byte]`, not a `ByteString`, in which case you should be able to create a `HttpBody` the same way you do in the other cases.  I'll create an issue for this, but def out of scope for this PR.